bundles/binding/org.openhab.binding.smlreader/src/main/java/org/openhab/binding/smlreader/internal/SmlReaderBinding.java

```diff
@@ -0,0 +1,288 @@
+				
+				if(obisValue == null) {
+					logger.debug("There is no value for id " + configuredItemObis + " actually.");
+					continue;
+				}
+				
+				if(configuredItemTransformationServiceType == null || configuredItemTransformationServiceType.isEmpty()) {
+					obisValueTransformed = obisValue;
+				} else { 
+					try {
+						TransformationService transformationService = 
+							TransformationHelper.getTransformationService(SmlReaderActivator.getContext(), configuredItemTransformationServiceType);
+						
+						if (transformationService != null) {
+							obisValueTransformed = transformationService.transform(configuredItemTransformationServiceFunc, obisValue);
+						} else {
+							obisValueTransformed = obisValue;
+						}
+					}
+					catch (TransformationException te) {
+						logger.error("transformation throws exception [transformation=" + configuredItemTransformationServiceType + ", obisValue=" + obisValue + "]", te);
+					
+						// in case of an error we return the response without any
+						// transformation
+						obisValueTransformed = obisValue;
+					}
+				}
+				
+				Class<? extends Item> itemType = provider.getItemType(itemName);
+
+				if(itemType.isAssignableFrom(NumberItem.class)){
+					eventPublisher.postUpdate(itemName, DecimalType.valueOf(obisValueTransformed));
+				} else if(itemType.isAssignableFrom(StringType.class)) {
+					eventPublisher.postUpdate(itemName, StringType.valueOf(obisValueTransformed));
+				}
+			}
+		}
+	}
+		
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void updated(Dictionary<String, ?> config) throws ConfigurationException {
+		if (config != null) {
+			// to override the default refresh interval one has to add a
+			// parameter to openhab.cfg like
+			// <bindingName>:refresh=<intervalInMs>
+			if (StringUtils.isNotBlank((String) config.get("refresh"))) {
+				refreshInterval = Long.parseLong((String) config.get("refresh"));
+			}
+			
+			setProperlyConfigured(true);
+		}
+		
+		Set<String> configuredDevices = getConfiguredDevices(config);
+		
+		for(String configuredDevice : configuredDevices){
+			String value = (String) config.get(configuredDevice + "." + SmlReaderConstants.Configuration.CONFIGURATION_KEY_CONNECTION_TYPE);
+			ConnectionType connectionType = ConnectionType.notConfigured;
+			
+			try{
+				 connectionType = ConnectionType.valueOf(value);
+			}catch(IllegalArgumentException | NullPointerException e){
+				logger.error("The connection type of device '" + configuredDevice + "# is invalid.");
+				setProperlyConfigured(false);
+				break;
+			}
+				
+			if(connectionType == ConnectionType.serial){
+				createFromSerialConfigurationSettings(configuredDevice, config);
+			} else if(connectionType == ConnectionType.tsap){
+				createFromTsapConfigurationSettings(configuredDevice, config);
+			} else{
+				logger.info("There is no device configured to read SML messages from.");
+				setProperlyConfigured(false);
```

Comment:
better move setProperlyConfigured to be the last statement of this method and rather set a local boolean meanwhile